server/src/actors/server/message.rs

```diff
@@ -42,36 +48,71 @@ impl Handler<Reconnect> for Server {
     fn handle(&mut self, msg: Reconnect, _ctx: &mut Context<Self>) -> Self::Result {
         let player_info = self.players.get(&msg.session_id);
         match player_info {
-            Some(player) => {
-                player.do_send(player::message::Connected {
-                    api_client: msg.api_client,
-                    session_id: msg.session_id,
-                });
-            }
-            None => {
-                msg.api_client.do_send(outbound_message::ConnectionFailed {
-                    reason: "Session not found".to_string(),
-                });
-            }
+            Some(player) => player.addr.do_send(player::message::Connected {
+                api_client: msg.api_client,
+            }),
+            None => msg.api_client.do_send(outbound_message::ConnectionFailed {
+                reason: "Session not found".to_string(),
+            }),
         }
     }
 }
 
+#[derive(Message)]
+#[rtype(result = "()")]
+pub struct DestroySession {
+    pub session_id: Uuid,
+}
+
+impl Handler<DestroySession> for Server {
+    type Result = ();
+
+    fn handle(&mut self, msg: DestroySession, _ctx: &mut Context<Self>) -> Self::Result {
+        debug!(
+            "Destorying session {}. Username: {}",
+            msg.session_id,
+            self.players
+                .get(&msg.session_id)
+                .map_or("N/A", |player_info| { player_info.username.as_str() })
+        );
+        self.players.remove(&msg.session_id);
+    }
+}
+
 #[derive(Message, Clone)]
 #[rtype(result = "()")]
 pub struct GlobalChatBroadcast {
-    pub timestamp: DateTime<Utc>,
-    pub system_msg: bool,
-    pub username: String,
-    pub message: String,
+    timestamp: DateTime<Utc>,
+    system_msg: bool,
+    username: String,
+    message: String,
+}
+
+impl GlobalChatBroadcast {
+    pub fn system_message(text: String) -> Self {
+        Self {
+            timestamp: Utc::now(),
+            system_msg: true,
+            username: "System".to_string(),
+            message: text,
+        }
+    }
+    pub fn user_message(username: String, text: String) -> Self {
```

Comment:
This copies the data, right? I'm looking here mainly at `text` - if it gets to a moderate size and this is called frequently copying might be inefficient. Perhaps you could check out some `Cow` stuff or just refs here...although this might get a bit tricky because of the async execution and the fact you want to keep the msgs immutable and owning the data :thinking: 

On the other hand, msgs probably won't be that long for this to make a difference.. :)

Reply:
I believe this should move the data, since the text is moved here (or copied at call point, but `text` is owned by the function), so everything should be just move. Only `.clone()` duplicates the data

Reply:
Ah, you're right, it's not `Copy` so there is no implicit copying, so it's moved here..mb